### PR TITLE
Added direct/indirect/unknown enum for IsDependency and update deps.dev collector/parser

### PR DESCRIPTION
**Pull request by pxp928**
Address Issue #769 

Closes https://github.com/guacsec/guac/issues/748

- Added direct/indirect/unknown enum for IsDependency
- [Deps.dev Parser] Updated parser for IsDependency to include direct/indirect/unknown info
- [Deps.dev collector] Updated parsing for [deps.dev](http://deps.dev/) to include structure of the tree
- [Deps.dev collector] enabled Collectsub identifier strings
- Removes scorecard compare from unit tests to ensure the CI doesnt break each time it changes in deps.dev (fixes https://github.com/guacsec/guac/issues/740)

### CODE REVIEW

File: pkg/ingestor/parser/spdx/parse_spdx.go [hunk 1]
```diff
@@ -183,6 +183,8 @@ func (s *spdxParser) GetPredicates(ctx context.Context) *assembler.IngestPredica
 
 	toplevel := s.getPackageElement("DOCUMENT")
 	// adding top level package edge manually for all depends on package
+	// TODO: This is not based on the relationship so that can be inaccurate (can capture both direct and in-direct)...Remove this and be done below by the s.spdxDoc.Relationships?
+	// see https://spdx.github.io/spdx-ntia-sbom-howto/ section 4.1.2. Additional subdependencies
 	if toplevel != nil {
 		preds.IsDependency = append(preds.IsDependency, common.CreateTopLevelIsDeps(toplevel[0], s.packagePackages, s.filePackages, "top-level package GUAC heuristic connecting to each file/package")...)
```
Comment (pxp928) [inline pkg/ingestor/parser/spdx/parse_spdx.go:189]: Based on the direct/indirect information we have, appending all packages to the top level seems like a false statement...

File: pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go [hunk 2]
```diff
@@ -181,6 +181,8 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 
 	toplevel := c.getPackageElement(string(c.cdxBom.Metadata.Component.BOMRef))
 	// adding top level package edge manually for all depends on package
+	// TODO: This is not based on the relationship so that can be inaccurate (can capture both direct and in-direct)...Remove this and be done below by the *c.cdxBom.Dependencies?
+	// see https://github.com/CycloneDX/specification/issues/33
 	if toplevel != nil {
 		preds.IsDependency = append(preds.IsDependency, common.CreateTopLevelIsDeps(toplevel[0], c.packagePackages, nil, "top-level package GUAC heuristic connecting to each file/package")...)
```
Comment (pxp928) [inline pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go:187]: similar comment to the SPDX one (appending to top level without relationship)

File: pkg/ingestor/parser/spdx/parse_spdx.go (hunk 1 above)
Comment (pxp928) [reply pkg/ingestor/parser/spdx/parse_spdx.go]: changed to `UKNOWN` for dependency type...

**Review by mihaimaruseac — APPROVED** [review]

**Comment by pxp928** [comment]
Ready for review

**Review by trmiller — APPROVED** [review]

**Pull request merged**